Realm/Realm/RealmConfiguration.cs

```diff
@@ -150,5 +179,15 @@ internal override Realm CreateRealm(RealmSchema schema)
             srHandle.SetHandle(srPtr);
             return new Realm(srHandle, this, schema);
         }
+
+        [NativeCallback(typeof(ShouldCompactCallback))]
+        private static bool ShouldCompactOnLaunchCallback(IntPtr configPtr, ulong totalSize, ulong dataSize)
+        {
+            var handle = GCHandle.FromIntPtr(configPtr);
+            var config = (RealmConfiguration)handle.Target;
+            var result = config.ShouldCompactOnLaunch(totalSize, dataSize);
```

Comment:
Maybe use a `try/finally` so that we always free the handle even if the user's delegate throws?